tests/src/Integration/Action/EntityCreateTest.php

```diff
@@ -0,0 +1,151 @@
+<?php
+
+/**
+ * @file
+ * Contains \Drupal\Tests\rules\Integration\Action\EntityCreateTest.
+ */
+
+namespace Drupal\Tests\rules\Integration\Action;
+
+use Drupal\Core\Entity\ContentEntityType;
+use Drupal\Tests\rules\Integration\RulesEntityIntegrationTestBase;
+
+/**
+ * @coversDefaultClass \Drupal\rules\Plugin\Action\EntityCreate
+ * @group rules_action
+ */
+class EntityCreateTest extends RulesEntityIntegrationTestBase {
+
+  /**
+   * A constant that will be used instead of an entity.
+   */
+  const ENTITY_REPLACEMENT = 'This is a fake entity';
+
+  /**
+   * The action to be tested.
+   *
+   * @var \Drupal\rules\Core\RulesActionInterface
+   */
+  protected $action;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    // Prepare mocked bundle field definition. This is needed because
+    // EntityCreateDeriver adds required contexs for required fields, and
+    // assumes that the bundle field is required.
+    $this->bundleFieldDefinition = $this->getMockBuilder('Drupal\Core\Field\BaseFieldDefinition')
+      ->disableOriginalConstructor()
+      ->getMock();
+
+    // The next methods are mocked because EntityCreateDeriver executes them.
+
+    $this->bundleFieldDefinition
+      ->expects($this->once())
+      ->method('getCardinality')
+      ->willReturn(1);
+
+    $this->bundleFieldDefinition
+      ->expects($this->once())
+      ->method('getType')
+      ->willReturn('string');
+
+    $this->bundleFieldDefinition
+      ->expects($this->once())
+      ->method('getLabel')
+      ->willReturn('Bundle');
+
+    $this->bundleFieldDefinition
+      ->expects($this->once())
+      ->method('getDescription')
+      ->willReturn('Bundle mock description');
+
+    // Prepare an content entity type instance.
+    $this->entityType = new ContentEntityType([
+      'id' => 'test',
+      'label' => 'Test',
+      'entity_keys' => [
+        'bundle' => 'bundle',
+      ],
+    ]);
+
+    // Prepare mocked entity storage.
+    $this->entityTypeStorage = $this->getMockBuilder('Drupal\Core\Entity\EntityStorageBase')
```

Comment:
Any reason why we mock the concrete class instead of the interface? Usually we should mock the interface if possible.